### PR TITLE
--docker-options flag added

### LINKED ISSUE #116: Add docker run options for appsody run
**Is your feature request related to a problem? Please describe.**
In order to enable more complex development environments, developers need the possibility to add more options for the `docker run` that is done internally. Setups that, for example, require more complex networking, would need to specify a `--network`. In order to be more flexible, we should allow to specify additional run arguments.

**Describe the solution you'd like**
The ability to specify additional docker run arguments, such as `--network`, `--memory`, `--privileged`, and potentially others.

### PR DESCRIPTION
**Pull request by kewegner**
Fixes #116 

Adds the --docker-options flag to allow users to send options to docker command (run, build) via the appsody command line.

### CODE REVIEW

File: cmd/build.go [hunk 1]
```diff
@@ -16,11 +16,30 @@ package cmd
...
 	"path/filepath"
+	"regexp"
+	"strings"
 
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
+func checkDockerBuildOptions(options []string) error {
+	buildOptionsTest := "(^((-t)|(--tag)|(-f)|(--file))(=?$)|(=.*))"
```
Comment (chilanti) [inline cmd/build.go]: I am not sure we should try and match for `=`.
Example: `docker build --file Dockerfile --tag mytag` would result in no matches for `--file Dockerfile --tag mytag`) - or am I confused?

Comment (chilanti) [reply cmd/build.go]: I did some testing - and this seems to yield matches for the blacklisted options: `(^((-t )|(--tag )|(-f )|(--file ))(.*))`

File: cmd/dev_common.go [hunk 2]
```diff
@@ -38,10 +38,25 @@ var depsVolumeName string
...
 var publishAllPorts bool
 var dockerNetwork string
-
+var dockerOptions string
 var nameFlags *flag.FlagSet
 var commonFlags *flag.FlagSet
 
+func checkDockerRunOptions(options []string) error {
+	fmt.Println("testing docker options", options)
+	runOptionsTest := "(^((-p)|(--publish)|(--publish-all)|(-P)|(-u)|(--user)|(--name)|(--network)|(-t)|(--tty)|(--rm)|(--entrypoint)|(-v)|(--volume)|(-e)|(--env))(=?$)|(=.*))"
```
Comment (chilanti) [inline cmd/dev_common.go]: Same as above?

File: cmd/build.go (hunk 1 above)
Comment (kewegner) [reply cmd/build.go]: The reason the code works is the string is split. and each individual piece is tested, not the whole string at once.

Comment (chilanti) [reply cmd/build.go]: Sorry - I had missed that.

**Pull request merged**